Convert zperf to use socket services API

jukkar opened the pull request:
I converted the zperf library to use socket services API in order to save memory. After the conversion, in my test setup (qemu_x86) 2836 bytes (ram) were saved.

For the zperf library and sample, the savings look like this:
```
main zperf rom_report 13602 ram_report 10120
pr   zperf rom_report 14081 ram_report  6184
                      =====            =====
                       -479             3936
```
So slight increasing for ROM usage, but quite substantial savings for RAM. The socket service implementation needs also memory so the total savings are a bit smaller.

jukkar committed:
net: shell: Do not crash if no sockets are found

The "net sockets" command was not checking if there is any
sockets in the system before trying to access them.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: shell: Make the thread name longer for sockets command

The thread name output field was a bit too short in "net sockets"
command, so make it 25 char long.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: zperf: Convert TCP receiver to use socket services

Use socket services API for TCP receiver.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
samples: net: zperf: Increase the polled fd count

We need more polled fd count so that socket services work ok.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: socket: Allow user to tweak service dispatcher thread priority

User is able to tweak the socket service dispatcher thread
priority in order to get better performance from the system
if needed. By default the dispatcher thread runs in lowest
application thread priority (K_LOWEST_APPLICATION_THREAD_PRIO).

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar commented:
Added a way to tweak the socket service dispatcher thread priority, this way it is possible to get best performance out of the stack.

rlubos commented on subsys/net/lib/zperf/zperf_tcp_receiver.c:
```diff
@@ -99,6 +96,139 @@ static void tcp_received(const struct sockaddr *addr, size_t datalen)
+		}
+
+		for (; i < SOCK_ID_MAX; i++) {
+			if (fds[i].fd == pev->event.fd) {
```
Would it be possible that we won't find FD on the list? Just wonder if we should check if `i == SOCK_ID_MAX` after the loop, and possibly handle it as an error?

jukkar replied:
Good point, there was a check in the listen socket branch but the check was missing here, I will add one.

rlubos reviewed (commented):
Looks good, just one question.

jukkar committed:
net: zperf: Convert UDP receiver to use socket services

Use socket services API for UDP receiver.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: socket: Change the printf modifier to print size_t correctly

The argument is size_t, so change the printf modifier to %zd
to avoid warning prints.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar commented:
Fixed the CI warning

pdgendt commented on subsys/net/lib/sockets/Kconfig:
```diff
@@ -86,6 +86,27 @@ config NET_SOCKETS_SERVICE
+	help
+	  Set the priority of the socket service dispatcher thread. This handler
+	  polls the sockets and either places the triggered socket to work queue
+	  for asynchrohous handlers, or calls the user supplied callback directly
```
```suggestion
	  for asynchronous handlers, or calls the user supplied callback directly
```
